verification/curator-service/ui/cypress/integration/utils/enterSource.ts

```diff
@@ -1,8 +1,16 @@
-export default function enterSource(url: string): void {
-    cy.get('div[data-testid="caseReference"]').type(`${url}{downarrow}{enter}`);
-    cy.get('input[id="name"]').type('New source');
-    cy.server();
-    cy.route('POST', '/api/sources').as('addSource');
-    cy.get('button[data-testid="sourceAdd"').click();
-    cy.wait('@addSource');
+export default function enterSource(url: string, existingSource = false): void {
+    if (existingSource) {
+        cy.get('div[data-testid="caseReference"]').type(
+            `${url}{downarrow}{enter}`,
+        );
+    } else {
+        cy.get('div[data-testid="caseReference"]').type(
+            `${url}{downarrow}{enter}`,
+        );
+        cy.get('input[id="name"]').type('New source');
```

Comment:
Do we need unique names across URLs?

Reply:
Nope, the name doesn't need to be unique (which is reasonable, IMO). What we'll likely want to implement, though, is a safeguard against duplicate URLs. I'll chat with Kelly about this next week -- I think ensuring the uniqueness at the data layer makes sense.